### PR TITLE
Final Polish translations

### PR DESCRIPTION
**Pull request by pauldim**
Final Polish translations. This PR is co-created with scr0llbaer9000. We can do the rest of the shortcomings later.

### CODE REVIEW

**Comment by scr0llbaer9000** [comment]
Did not come around for a final round of checks / fixes:

- All references to roles in `eggi_screenplays_adaptations.xml` must be fixed (wrong quotation marks messed up by translation tool) (had noticed that, but forgot).

- `database_programmes.xml` has a `[2|Nicka]a` that must be fixed to `${.self:"cast":2:"nickname"}a` (validator kinda finds that by coincidence)

- variable options count mismatches with the other languages in `database_scripts.xml` in lines 394, 939, 970, 980, 1021

- variable options count mismatch in `scr0llbaer_news.xml in lines` 1336 f (forgot some `|` characters at the end)

- variable options count mismatch in `scr0llbaer_screenplays_adaptations.xml` in lines 6292, 7356, 7694, 9517

- property name got errouneously tanslated in `scr0llbaer_screenplays_adaptations.xml` line 1694 (must be `"firstname"` instead of `"imię"`)

That's all I have for now.

btw validator doesn't update in my eclipse @nittka , is that problem on your side? So it's still a bit hard to wade through everything.

**Comment by nittka** [comment]
I will review the changes and give additional feedback. After the merge, my suggetion would be for pauldim to create a backup-branch of his current master (if wanted) and do a hard-reset + force-push on the official master (cannot be done in the browser, but must be done from a client). In the standard case, the fork's master should be kept in synch by pulling/fast forward merges rather than merge commits. Contributions/on changes should be done on a branch and not the master.

@scr0llbaer9000 : There are still some PRs open that have impact on the editor. I wanted to know if they are merged or rejected before I release a new editor version.

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:1694]: parameter firstname - not translated

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:6292]: 5 alternatives here, 6 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:7356]: 12 alternatives here, 13 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:7694]: 4 alternatives here, 5 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:8326]: no role references here?

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:9517]: 14 alternatives here, 15 in the other languages

**Review by nittka — CHANGES_REQUESTED** [review]
I still have to go through the changes in the main programmes file (old cast reference in line 84566 [2|Nick...).
Please fix the formal issues first. Only after that is done, it makes sense to run the test script that tries to find unresolvable references/variables.

**Comment by nittka** [comment]
In programmes.xml very many original titles for French/Italien etc. have been removed. (I compared the files from master and this PR with all Polish tags removed - ideally they should be largely identical).

File: res/database/Default/user/eggi_screenplays_adaptations.xml
Comment (pauldim) [reply res/database/Default/user/eggi_screenplays_adaptations.xml]: Done

File: res/database/Default/database_news.xml
Comment (pauldim) [reply res/database/Default/database_news.xml:8574]: Corrected

File: res/database/Default/database_news.xml
Comment (pauldim) [reply res/database/Default/database_news.xml:8114]: There is so much to look at and so many mistakes. Corrected.

File: res/database/Default/database_scripts.xml
Comment (pauldim) [reply res/database/Default/database_scripts.xml:745]: Done

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml
Comment (pauldim) [reply res/database/Default/user/scr0llbaer_screenplays_adaptations.xml]: Grammar difficulties :)

File: res/database/Default/user/ariksan.xml
Comment (pauldim) [reply res/database/Default/user/ariksan.xml:7]: <pl>${title} w ${randomcity}</pl> - There must be a Polish version

File: res/database/Default/database_ads.xml
Comment (pauldim) [reply res/database/Default/database_ads.xml:270]: Media Markt in Poland. Great , that we have this functionality of original ads. I just noticed it.

**Review by nittka — CHANGES_REQUESTED** [review]
In case you are interested where Polish translations (title) are still missing (probably last minute additions in one of the recent commits):
* scr0llbaer.xml line 1271 and following: d26ee4f9-1b61-4b7a-926a-e462256b127b and episodes
* scr0llbaer.xml line 1354: d664f243-15d6-4645-886e-27a3e8cf8533
* database_ads.xml line: 360b9ee1-9e2e-469d-b61e-fa8a69cbf560 (set brand explicitly)
* database_programmes.xml line 7622: 216d7e50-2893-4a2e-973a-4f55a3ab81b0
* database_programmes.xml line 53814: 6543de67-43ce-4a32-ab3f-c7c1dbbae34a

File: res/database/Default/user/ariksan.xml
Comment (pauldim) [reply res/database/Default/user/ariksan.xml]: Like this:
<adjective>
					<de>witzige|aufgeweckte|gescheite|kluge|lustige</de>
					<en>Witty|Sharp|Clever|Smart|Funny</en>
					<pl>Dowcipn|Ostr|Sprytn|Inteligentn|Zabawn</pl>
				</adjective>
				<adjective_m>
					<pl>${adjective}y</pl>
				</adjective_m>
				<adjective_w>
					<pl>${adjective}a</pl>
				</adjective_w>
Corrected

**Comment by nittka** [comment]
I started checking with the test script...
* database_programmes line 112010: you fixed the reference, but the square brackets are still present `[` and `]` should be removed
* ariksan line 511: variable name ${animal2} was translated
* database_scripts line 1104: variable ${who_${i}} should be replaced by ${who}
* scr0llbar_news line 1362: the number of entries within the csv tvtowermeta_csv does not match. It looks like one pair of translations is missing (8 entries instead of 10). Also, are the white spaces after the semicolon intended? They will be present in the resulting text.
* database_news line 3981: again too few csv elements; I guess, the comma after Rio de Janeiro should be a semicolon. Note also that in the alternatives of this entry, there is always a white space in the second entry except for London (first alternative)

**Comment by nittka** [comment]
database_news line 7000: one csv entry missing - i guess the comma after heavy metalovy should be a semicolon

**Comment by pauldim** [comment]
`<scripttemplate product="1" licence_type="1" guid="443fa70e-450c-4085-a4de-c41b23e295ef" creator="9551" comment="Description largely by ChatGPT">
			<title>
				<de>Sekretärinnen-Bericht 3. Teil - Stenografie, Streiche &amp; Sinnlichkeit</de>
				<en>Secretaries Report Part 3 - Shorthand, Stunts &amp; Sensuality</en>
				<pl>Raport sekretarek, Część 3 - Skróty, akrobacje i zmysłowość</pl>
			</title>
			<description>
				<de>Jetzt wird's noch bunter: ${.self:"role":1:"fullname"} wird zur Meisterin der Stenografie und anderer Quickies, Fax und Fon klingeln im Rhythmus der Aktion und ${.self:"role":2:"fullname"}s klappernder Schreibmaschine... Inmitten biederer Bürokratie brodelt feurige Fantasie.</de>
				<en>Now it all gets extra sparkly: ${.self:"role":1:"fullname"} becomes a master of shorthand and other quickies, fax and phone ring to the rhythm of the action and ${.self:"role":2:"fullname"}'s clattering typewriter.**,,** In the midst of bland bureaucracy, fiery fantasies are fizzing.</en>`

There should be two commas in the English version?

**Comment by nittka** [comment]
No, I think the two commas should be dots - as in German. typewriter... In the midst

**Comment by pauldim** [comment]
I think everything has been corrected. I don't think I missed anything.

**Comment by scr0llbaer9000** [comment]
> After the merge, my suggetion would be for pauldim to create a backup-branch of his current master (if wanted) and do a hard-reset + force-push

...or just delete the TVTower fork and fork TVTower again. And then create a *branch* for your own experiments.

**Comment by nittka** [comment]
Test script and editor validation look fine. I'll merge tomorrow if I don't hear any objections.

**Comment by scr0llbaer9000** [comment]
The texts for the new blockbuster achievements in [database_achievements.xml](https://github.com/TVTower/TVTower/blob/master/res/database/Default/database_achievements.xml) are still untranslated.

Only very few lines that could be done in this PR, then that will be all as far as I can see.

**Comment by pauldim** [comment]
> @pauldim : please let me know, if you want to address the blockbuster achievement.

I didn't notice that comment. I've already added those translations.

**Comment by nittka** [comment]
Almost there. When working on improvements on the editor validation two subtle issues came up that the current validation and the test script missed (note that this script cannot guarantee 100% coverage of all possible alternatives, so even in combination editor and test script are not guaranteed to find all issues).

@scr0llbaer9000 and @pauldim please let me know if you want to correct those issues in this PR or separately (Polish translation is affected for one of them)

* trivial issue: scr0llbaer_news line 416 (733f6d6a-4021-45c0-8ace-7768f9a922f0) - in German the last alternative ends with pipe (|), i.e. there is another unintended empty alternative; remove the pipe
* less trivial: scr0llbar_news lines 630 and following (8ac54969-a412-4c23-98c1-a3b2f449605d) - the csv data alternatives 5 (all languages) and 6 (en and pl) of the variable what_csv are not consistent with the rest. All other entries have 3 data points (2x semicolon) but 5 and 6 have only 2 data points (1x semicolon). But the third datapoint is never referenced (...what_csv:2 cannot be found), so the texts there will never be used. Note that there is no support for escaping at the moment.

**Comment by scr0llbaer9000** [comment]
> trivial issue

not so trivial? or why didn't the validator find it before?

> scr0llbar_news lines 630 and following

ah yeah, these are "real" semicolons. Was wondering there might be instances of those. Originally used them because of casing. Can change to full stops now.

I'll fix these when you merge.

**Comment by nittka** [comment]
> or why didn't the validator find it before?

The bug simply never surfaced. If TVTower determines the used alternative based on English (no idea), the additional German one will never cause errors. If the 100 interations of the test script never chooses the last German one... With the validation working for the rest of the cases, there was no reason to check the code.

New editor version is available.

**Pull request merged**
